### PR TITLE
hook up checksums to trs

### PR DESCRIPTION
**Pull request by NatalieEO**
https://ucsc-cgl.atlassian.net/browse/SEAB-845

### CODE REVIEW

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dockstore/dockstore/pullRequest?prid=5078825&bid=16706991)

File: dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java [hunk 1]
```diff
@@ -155,7 +156,12 @@ public static Tool convertEntryToTool(Entry<?, ?> container, DockstoreWebservice
             //TODO: would hook up identified tools that form workflows here
             toolVersion.setIncludedApps(MoreObjects.firstNonNull(toolVersion.getIncludedApps(), Lists.newArrayList()));
             //TODO: hook up snapshot and checksum behaviour here
-            toolVersion.setIsProduction(false);
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java]: I think you should restore this line, change it to `toolVersion.setIsProduction(version.isFrozen())`, then get rid of the else on line 162.

File: dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java [hunk 2]
```diff
@@ -155,7 +156,12 @@ public static Tool convertEntryToTool(Entry<?, ?> container, DockstoreWebservice
...
+            if (version.isFrozen()) {
+                List<ImageData> trsImages = processImageDataForWorkflowVersions(version, toolVersion);
+                toolVersion.setImages(MoreObjects.firstNonNull(trsImages, Lists.newArrayList()));
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java]: It doesn't look like procesImageDataforWorkflowVersions returns a null (nor should it), so this line is unnecessary.

File: dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java [hunk 3]
```diff
@@ -230,6 +236,32 @@ public static Tool convertEntryToTool(Entry<?, ?> container, DockstoreWebservice
         return tool;
     }
 
+    private static List<ImageData> processImageDataForWorkflowVersions(final Version version, final ToolVersion toolVersion) {
+        toolVersion.setIsProduction(true);
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java]: I would get rid of this (see above comment). This is a side-effect and generally not a recommended practice -- I would expect a method with a signature like this to return a List<ImageData>, not return a List and modify an object.

File: dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java [hunk 4]
```diff
@@ -230,6 +236,32 @@ public static Tool convertEntryToTool(Entry<?, ?> container, DockstoreWebservice
...
+        for (Image image : images) {
+            ImageData imageData = new ImageData();
+            imageData.setImageType(ImageType.DOCKER);
+            //TODO: For workflows, it's just quay for now, but that'll change. Should add column in image table to store registry.
+            imageData.setRegistryHost("quay.io");
+            imageData.setImageName(constructName(Arrays.asList(image.getRepository(), image.getTag())));
+            List<Checksum> trsChecksums = new ArrayList<>();
+            List<io.dockstore.webservice.core.Checksum> checksumList = image.getChecksums();
+
+            for (io.dockstore.webservice.core.Checksum checksum : checksumList) {
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java]: This is fine, but in the future you might want to look at using `map` instead as an alternative and slightly more functional way (fix not required). And also for the outer loop.

File: dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java [hunk 5]
```diff
@@ -230,6 +236,32 @@ public static Tool convertEntryToTool(Entry<?, ?> container, DockstoreWebservice
...
+    private static List<ImageData> processImageDataForWorkflowVersions(final Version version, final ToolVersion toolVersion) {
+        toolVersion.setIsProduction(true);
+        Set<Image> images = version.getImages();
+        List<ImageData> trsImages = new ArrayList<>();
+
+        for (Image image : images) {
+            ImageData imageData = new ImageData();
+            imageData.setImageType(ImageType.DOCKER);
+            //TODO: For workflows, it's just quay for now, but that'll change. Should add column in image table to store registry.
+            imageData.setRegistryHost("quay.io");
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java]: I know it's for now, but this sort of scares me if we get distracted and don't come back to it. Is there a ticket for this?

Comment (NatalieEO) [reply dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java]: It was something I realized while doing this PR, so just made an issue here https://github.com/dockstore/dockstore/issues/3275

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dockstore/dockstore/pullRequest?prid=5078825&bid=16706991)

File: dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java [hunk 6]
```diff
@@ -867,6 +877,20 @@ public void testGettingImagesFromQuay() {
         WorkflowVersion versionWithDuplicateImages = snapshotWorkflowVersion(workflowsApi, "dockstore-testing/zhanghj-8555114", "/main.cwl", "1.0");
         assertEquals("Should have grabbed 3 images", 3, versionWithDuplicateImages.getImages().size());
         verifyChecksumsAreSaved(versionWithDuplicateImages);
+        versions = ga4Ghv20Api.toolsIdVersionsGet("#workflow/github.com/dockstore-testing/zhanghj-8555114");
+        testTRSConversion(versions, "1.0", 3);
+    }
+
+    private void testTRSConversion(final List<ToolVersion> versions, String snapShottedVersionName, final int numImages) {
+        for (ToolVersion trsVersion : versions) {
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java]: Will pass if something horribly breaks and there are no versions, should have an assert that this is working as expected and that there are versions (and if I understand the intent, that that one of them should be the snapShottedVersionName)

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dockstore/dockstore/pullRequest?prid=5078825&bid=16706991)

**Pull request merged**